solr/core/src/java/org/apache/solr/update/processor/ParseDateFieldUpdateProcessorFactory.java

```diff
@@ -115,9 +123,10 @@ protected Object mutateValue(Object srcVal) {
           for (Map.Entry<String,DateTimeFormatter> format : formats.entrySet()) {
             DateTimeFormatter parser = format.getValue();
             try {
-              DateTime dateTime = parser.parseDateTime(srcStringVal);
-              return dateTime.withZone(DateTimeZone.UTC).toDate();
-            } catch (IllegalArgumentException e) {
+              TemporalAccessor parsedTemporalDate = parser.parseBest(srcStringVal, OffsetDateTime::from,
```

Comment:
IMO we should strive for better performance in URPs.
I had a look at your patch, and used it as a building block to make another parsing method.
Perhaps we should have a more universal parsing method in a more global scope to avoid code duplication.

Reply:
I determined the functionality in the "extraction" module should simply go away in lieu of this URP, so there will be no duplication.

Reply:
Ok, would this be part of another ticket?

Reply:
SOLR-12591 is next (docs + tests), then removal in SOLR-12593

Reply:
Alright.
Hopefully this one is close to being merged. Any other problems you for see here?